src/Entity/Traits/ActivityPubActorTrait.php

```diff
@@ -79,12 +79,24 @@ public function getPublicKey(): ?string
         return $this->publicKey;
     }
 
-    public function rotatePrivateKey(): void
+    public function rotatePrivateKey(bool $revert = false): void
     {
-        $this->oldPrivateKey = $this->privateKey;
-        $this->oldPublicKey = $this->publicKey;
-        // set new private and public key
-        KeysGenerator::generate($this);
+        if (!$revert) {
+            $this->oldPrivateKey = $this->privateKey;
+            $this->oldPublicKey = $this->publicKey;
+            // set new private and public key
+            KeysGenerator::generate($this);
+        } else {
+            if (null === $this->oldPrivateKey || null === $this->oldPublicKey) {
+                throw new \InvalidArgumentException('you cannot revert if there is no old key');
```

Comment:
`oldPrivateKey` & `oldPublicKey` now only lives in memory right? 

So a simple PHP-FPM restart will get rid of the old keys. So the message should be more about that you can't revert the keys in case you restarted PHP or the server? Since after all the data will be lost?